llvm/lib/Support/KnownFPClass.cpp

```diff
@@ -137,6 +137,30 @@ KnownFPClass KnownFPClass::canonicalize(const KnownFPClass &KnownSrc,
   return Known;
 }
 
+KnownFPClass KnownFPClass::exp(const KnownFPClass &KnownSrc) {
+  KnownFPClass Known;
+  Known.knownNot(fcNegative);
+
+  if (KnownSrc.isKnownNeverNaN()) {
+    Known.knownNot(fcNan);
+    Known.signBitMustBeZero();
```

Comment:
knownNot(fcNagative) and knownNot(fcNan) implies this property. It will be updated in `Known.knownNot(fcNan);`.